Proposal: initial project setup

nicolas-goudry opened the pull request:
This PR setup the repository with the following:

* basic git stuff
  * gitignore
  * husky: for version controlled hooks
* enforce commit style: uses commitlint with config-conventional
* enforce code style: uses standard with comma-dangle custom rule (for easier reviews)
* add Github issue templates for:
  * feature requests
  * issues
* add main readme file with some todo
* add contributing guide with code of conduct

Let me know what you think about this!

cc @brochington

brochington commented:
All of this so far looks good to me. Thank you for doing it. :shipit: 

I would like propose using The Typescript “language” (probably not the actual tsc compiler) for this project. I’ve found it invaluable. I can set it up.

What are preferences for testing? Jest?

nicolas-goudry commented:
Never used Typescript, so I could use some help, but I’m not against it, go ahead :wink: 

Regarding testing, I’m more of a backend guy, so I never used Jest, but you seems to be more front-end than me, so I’ll let you chose!

Should we wait for this to be merged, or maybe you could push your work on my fork so it’s added to this PR?

brochington commented:
Copy on using Typescript. Happy to help you with it, if needed. 

Yeah, I do mostly FE work nowadays. Jest seems to be a good standard testing library, So I can add that too. 

I don't mind waiting a bit for the PR to be merged, as long as it's not going to take like...weeks. @lipis Any guidance on moving forward with things like PRs?

nicolas-goudry commented:
I heard about [testing library](https://testing-library.com/) getting much traction lately, ever heard of it?

lipis commented:
Hey didn't forget you.. I just didn't get the chance! Will get back to you soon.

lipis commented:
This looks great few things:

- Let's start simple and not include the templates for issues/PRs
- Same for code of conduct.. we're missing one for the core, and once we'll get it we'll use it on all projects (most likely)
- For Prettier user [`@excalidraw/prettier-config`](https://www.npmjs.com/package/@excalidraw/prettier-config) just released :)
- For ESlint we want to make something similar to have the same for all projects as we grow (https://github.com/excalidraw/eslint-config), but it's still under construction :) Let's leave it for now but just a heads up.
- For commits we are using [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged). I never seen your setup, but let's fly with it and we might change it later.

lipis commented:
I also invited you @nicolas-goudry to the https://github.com/orgs/excalidraw/teams/svg-to-excalidraw.. anyone else that we need to get there?

nicolas-goudry commented:
I’ll make these adjustments tonight and merge this PR.

Duly noted for ESLint config, we’ll try to keep an eye on it, I’ll just raise an issue in this repo so we don’t forget.

I’m not familiar with lint-staged, I’ll give it a look. Husky is set up. I just added commitlint to enforce commits style.

Us two are the only ones working on this for now, so it’s ok!

Thanks for everything :wink:

lipis commented on .eslintignore:
```diff
@@ -0,0 +1,2 @@
+dist
+node_modules
```
end of line

lipis commented:
We also recently switched to Yarn..

nicolas-goudry commented:
@lipis I think I didn’t forget anything!

- removed code of conduct
- removed Github issue templates
- switched to @excalidraw/prettier-config (and ran it)
- setup lint-staged (with husky v5)
- fixed missing EOL
- switched to yarn
- setup Github Actions workflows

lipis commented on README.md:
````diff
@@ -1 +1,70 @@
+## :floppy_disk: Installation
+
+```bash
+npm install svg-to-excalidraw --save # or yarn add svg-to-excalidraw
````
yarn

lipis commented on README.md:
````diff
@@ -1 +1,70 @@
+npm run build # builds the project
+# or
+npm run build:watch # builds project whenever a file is updated.
+```
````
yarn

lipis commented on webpack.config.js:
```diff
@@ -0,0 +1,76 @@
+  //   libraryTarget: 'umd',
+  //   publicPath: '/static/',
+  //   umdNamedDefine: true,
+  // },
```
use it or remove it..

lipis commented on package.json:
```diff
@@ -0,0 +1,70 @@
+    "serve": "^11.3.2",
+    "typescript": "^4.1.5",
+    "webpack": "^5.23.0",
+    "webpack-cli": "^4.5.0"
```
pinned versions for less surprises..
